Issue#33 first migration

Arthurpanda opened the pull request:
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

mykytam committed:
Issue#34: Create User entity and configure Auth
Add spring security dependency
Add User model and Role enum
Add basic SecurityConfig file

isvyry committed:
Merge branch 'dev' into Issue#34-Create-User-entity-and-configure-Auth

isvyry committed:
Issue#34: Create User entity and configure Auth
Add @Autowired above constructor in the api.mustache template to prevent build failure;
Add user models and endpoints to the spec file;
Add initial SecurityConfig and user entity to configure Auth;
Add initial implementation of Authentication Provider;
Add UserRepository for accessing users in db.

isvyry committed:
Issue#34: Create User entity and configure Auth
Moved spring-starter-security dependency to general pom
Add password encryption

mykytam committed:
Issue#34: Create User entity and configure Auth
Update entities to use RBAC structure

isvyry committed:
Issue#34: Create User entity and configure Auth
Fix problem

Valera3107 committed:
Add liquibase db migrations

Commit:
wip

Commit:
wip

Commit:
do not use cookies

isvyry committed:
Issue#34: Create User entity and configure Auth
Removed Authentication provider and set UserDetailsService instead

isvyry committed:
Issue#34: Create User entity and configure Auth
Add locked, expired, credentialsExpired, enabled - boolean parameters to User entity

isvyry committed:
Issue#34: Create User entity and configure Auth
Add dev profile in case we need load initial data and use it for tests

isvyry committed:
Merge remote-tracking branch 'origin/dev' into Issue#34-Create-User-entity-and-configure-Auth-problem

# Conflicts:
#	home-data/src/main/resources/application.properties
#	home-open-api/src/main/resources/yaml/openapi.yaml

isvyry committed:
Issue#34: Create User entity and configure Auth
Add newlines at the end of file

isvyry committed:
Issue#34: Create User entity and configure Auth
Fix variable naming

isvyry committed:
Issue#34: Create User entity and configure Auth
Fix package naming
Rewrite getAuthorities() method in a stream way
Delete test service class
Disable dev-profile

Valera3107 committed:
Add some classes to module home-data-migration

Commit:
wip

isvyry committed:
Issue#34: Create User Entity and configure Auth
Delete Eager FetchType from entity
Fix LazyInitializationException in User Details

isvyry committed:
Issue#34: Create User Entity and configure Auth
Add newline at the end of file

Valera3107 committed:
Add some classes to module home-data-migration

isvyry committed:
Merge remote-tracking branch 'origin/dev' into Issue#34-Create-User-entity-and-configure-Auth-problem

# Conflicts:
#	home-data/src/main/resources/application.properties

Valera3107 committed:
Add some classes to module home-data-migration

Valera3107 committed:
Add some classes to module home-data-migration

Valera3107 committed:
Add some classes to module home-data-migration

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/HomeApplication.java:
```diff
@@ -7,5 +7,6 @@
 public class HomeApplication  {
     public static void main(String[] args) {
          SpringApplication.run(HomeApplication.class, args);
+
```
Why do we need this extra line?

kh0ma commented on home-data-migration/pom.xml:
```diff
@@ -12,5 +12,52 @@
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>4.0.0</version>
```
Can we introduce a property?

kh0ma commented on home-data-migration/pom.xml:
```diff
@@ -12,5 +12,52 @@
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
```
?

kh0ma commented on home-data-migration/pom.xml:
```diff
@@ -12,5 +12,52 @@
+            <plugin>
+                <groupId>org.liquibase</groupId>
+                <artifactId>liquibase-maven-plugin</artifactId>
+                <version>3.6.3</version>
```
use property

kh0ma commented on home-data-migration/pom.xml:
```diff
@@ -12,5 +12,52 @@
+                </dependencies>
+                <executions>
+                    <execution>
+                        <id>202011092145</id>
```
What is the magic number? Can you properly name this execution?

kh0ma commented on home-data-migration/src/main/java/Main.java:
```diff
@@ -0,0 +1,16 @@
+        String pass, user;
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please, enter the username");
+        user = scanner.nextLine();
```
Please do not use any prompts. We'll use this app in a non-tty environment so just take required data from arguments.

kh0ma commented on home-data-migration/src/main/java/LiquibaseUpdate.java:
```diff
@@ -0,0 +1,66 @@
+    private static final String pathname = "changelog.path";
+    private static final String changelog = "changelog.name";
+
+    public void runLiquibase(String user, String password) {
```
With a high probability URL of db will vary on the environments. Please use the default if not present approach and give a possibility to specify it.

kh0ma commented on home-data-migration/src/main/java/LiquibaseUpdate.java:
```diff
@@ -0,0 +1,66 @@
+//    Get value from migration.properties
+    public String getFromProperties(String value) {
+        String url = "";
+        try (InputStream input = new FileInputStream("home-data-migration\\src\\main\\resources\\migration.properties")) {
```
I don't think this URI will be present somewhere out of the project. Please specify one from the classpath.

kh0ma commented on home-data-migration/src/main/java/LiquibaseUpdate.java:
```diff
@@ -0,0 +1,66 @@
+            liquibase = new Liquibase(changelogName, new FileSystemResourceAccessor(new File(path)), database);
+            liquibase.update(new Contexts());
+        } catch (SQLException | LiquibaseException | ClassNotFoundException e) {
+            e.printStackTrace();
```
Please wrap it as runtime and rethrow or just declare in the method signature.

kh0ma commented on home-data-migration/src/main/java/LiquibaseUpdate.java:
```diff
@@ -0,0 +1,66 @@
+                    connection.rollback();
+                    connection.close();
+                } catch (SQLException e) {
+                    //nothing to do
```
I think this condition should be logged as error. Please add logging.

kh0ma commented on home-data-migration/src/main/java/Main.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.Scanner;
+
+public class Main {
```
Specify proper name for the class. Use proper package for the class.

kh0ma commented on home-data-migration/src/main/resources/db/changelog/v-1.0/001-changeset-users-table.xml:
```diff
@@ -0,0 +1,19 @@
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="001" author="Artur">
+        <sqlFile dbms="postgresql"
```
Please get rid of sql file. Use liquibase dsl.

kh0ma commented on home-data-migration/src/main/resources/db/changelog/v-1.0/db.changelog-v.1.0.xml:
```diff
@@ -0,0 +1,12 @@
+
+
+    <include file="001-changeset-users-table.xml" relativeToChangelogFile="true"/>
+<!--    <include file="002-changeset-house-table.xml" relativeToChangelogFile="true"/>-->
```
Why is it commented out and present in the review without proper comment?

kh0ma commented on home-data-migration/src/main/resources/migration.properties:
```diff
@@ -0,0 +1,4 @@
+url = jdbc:postgresql://localhost:5432/test
+changelog.path = home-data-migration\\src\\main\\resources\\db\\changelog
```
I think this path isn't relevant for using this app outside of the project sources. Please use the one from classpath.

kh0ma commented on home-data/pom.xml:
```diff
@@ -18,11 +18,18 @@
+                    <artifactId>postgresql</artifactId>
+                    <scope>runtime</scope>
+                </dependency>
+<!--        <dependency>-->
```
Why is it here?

kh0ma commented on home-data/src/main/resources/application.properties:
```diff
@@ -1,3 +1,16 @@
+#spring.liquibase.change-log=classpath:db/changelog/db.changelog-master.xml
+spring.jpa.hibernate.ddl-auto=validate
+spring.jpa.database=POSTGRESQL
+spring.jpa.generate-ddl = true
```
?

kh0ma commented on home-data/src/main/resources/application.properties:
```diff
@@ -1,3 +1,16 @@
+spring.datasource.password=${dbPass:1234}
+spring.datasource.driverClassName=org.postgresql.Driver
+
+#spring.datasource.url=${dbUrl:jdbc:mysql://localhost:3306/test?createDatabaseIfNotExist=true&serverTimezone=UTC}
```
?

kh0ma commented on pom.xml:
```diff
@@ -28,6 +28,11 @@
         <maven.compiler.plugin.version>3.8.1</maven.compiler.plugin.version>
+
+        <!-- Password and root name for liquibase-->
+        <db_USERNAME>postgres</db_USERNAME>
```
use proper case for property names

kh0ma commented on pom.xml:
```diff
@@ -28,6 +28,11 @@
+        <!-- Password and root name for liquibase-->
+        <db_USERNAME>postgres</db_USERNAME>
+        <db_PASSWORD>1234</db_PASSWORD>
+        <db_HOST>jdbc:postgresql://localhost:5432/test</db_HOST>
```
I think db.host doesn't represent the value of this property.

kh0ma commented on pom.xml:
```diff
@@ -72,6 +77,7 @@
             <plugin>
                 <groupId>org.codehaus.mojo</groupId>
                 <artifactId>flatten-maven-plugin</artifactId>
+                <version>1.2.5</version>
```
Introduce propertry.

isvyry committed:
Merge pull request #48 from ita-social-projects/Issue#34-Create-User-entity-and-configure-Auth-problem

Issue#34 create user entity and configure auth

mykytam committed:
Issue#35: Users CRUD Change User OpenApi Specification
Change DTOs naming,
Change operationIds naming,
Change interger type to long for Ids,
Change typography of user's tag.

mykytam committed:
Issue#35: Users CRUD Change User OpenApi Specification
Change DTOs naming

mykytam committed:
Merge pull request #68 from ita-social-projects/Issue#35-Users-CRUD-Change-User-OpenApi-Specification

Issue#35: Users CRUD Change User OpenApi Specification

nrevutska committed:
Set up CI with Azure Pipelines

[skip ci]

nrevutska committed:
Delete azure-pipelines.yml

Valera3107 committed:
Merge branch 'Issue#33-First-migration' into dev

# Conflicts:
#	home-data/src/main/resources/application.properties

Valera3107 committed:
Add fat jar file

Valera3107 committed:
Add fat jar file

mykytam commented on .github/PULL_REQUEST_TEMPLATE/develop.md:
```diff
@@ -1,7 +1,7 @@
 ## JIRA
 
-* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
+* [com.softserveinc.ita.homeproject.migrations.LiquibaseRunner JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
```
Why do we change PR template here?

mykytam commented on .github/PULL_REQUEST_TEMPLATE/master.md:
```diff
@@ -1,7 +1,7 @@
 ## JIRA
 
-* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
+* [com.softserveinc.ita.homeproject.migrations.LiquibaseRunner JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
```
Why do we change PR template here?

mykytam commented on home-data/src/main/resources/application.properties:
```diff
@@ -1,5 +1,3 @@
-spring.datasource.password=${DATASOURCE_PASSWORD:password}
-spring.jersey.type=filter
-server.port=${PORT:8080}
+spring.datasource.url=${dbUrl:jdbc:postgresql://localhost:5432/test}
+spring.datasource.username=${dbUsername:postgres}
+spring.datasource.password=${dbPass:54321}
```
Why did you wrote here other default values?

mykytam commented on home-data-migration/src/main/resources/db/changelog/v-1.0/001-create-users-table.sql:
```diff
@@ -0,0 +1,12 @@
+                            last_name varchar (255)  not null,
+                            login varchar (255) not null,
+                            password varchar (255)  not null,
+                            phone_number varchar (255) not null,
```
Do we really have `_phone_number_` in database? Isn't it should be called` contacts`?
